src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor({ health = 100, name }) {
+    this.health = health;
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(bittedAnimal) {
+    if (bittedAnimal instanceof Herbivore && !bittedAnimal.hidden) {
+      bittedAnimal.health -= 50;
+    }
+
+    if (bittedAnimal.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== bittedAnimal);
+    }
```

Comment:
The `bite` method should not perform the `Animal.alive` filtering if `bittedAnimal` is a Carnivore or is hidden. Move the health check and filtering logic inside the `if` statement that checks if `bittedAnimal` is an instance of `Herbivore` and not hidden.